etcdserver/etcdserverpb/raft_internal.proto

```diff
@@ -10,10 +10,17 @@ option (gogoproto.sizer_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 option (gogoproto.goproto_getters_all) = false;
 
+message RequestHeader {
+  uint64 ID = 1;
+  // username is a username that is associated with an auth token of gRPC connection
+  string username = 2;
+}
+
 // An InternalRaftRequest is the union of all requests which can be
 // sent via raft.
 message InternalRaftRequest {
-  uint64 ID = 1;
+  RequestHeader header = 1;
```

Comment:
let's do not break the existing proto. put request header at top with a higher field like 100?
for the request, we need to check the header.id field first if it does not exist then check id.

we can deprecate uint64 ID after we release 3.0 (we want people to be able to upgrade their cluster for beta to 3.0)